src/elements/Input/Input.js

```diff
@@ -1,74 +1,100 @@
-    // The element must be on the same side as the indicated class.
-    // We first determine the left/right classes for each type of child,
-    //   then we extract the children and place them on the correct side
-    //   of the input.
-    const isLeftAction = _.includes(className, 'left action')
-    const isRightAction = !isLeftAction && _.includes(className, 'action')
-    const isRightLabeled = _.includes(className, 'right labeled')
-    const isLeftLabeled = !isRightLabeled && _.includes(className, 'labeled')
-
-    const labelChildren = []
-    const actionChildren = []
-
-    Children.forEach(children, child => {
-      const isButton = child.type.name === 'Button'
-      const isDropdown = child.type.name === 'Dropdown'
-      // TODO: use child.type.name === 'Label' once Label component is merged.
-      const isLabel = _.isString(child.props.className) && !!child.props.className.match(/ui.*label$/)
-      const childIsAction = !isLabel && isButton || isDropdown
-
-      if (childIsAction) {
-        actionChildren.push(child)
-      } else if (isLabel) {
-        labelChildren.push(child)
-      }
-    })
-
-    const classes = classNames(
-      'sd-input',
-      'ui',
-      className,
-      'input'
-    )
-    const props = getUnhandledProps(Input, this.props)
-
-    return (
-      <div className={classes}>
-        {isLeftLabeled && labelChildren}
-        {isLeftAction && actionChildren}
-        <input {...props} type={type} />
-        {icon && <Icon className={icon} />}
-        {isRightLabeled && labelChildren}
-        {isRightAction && actionChildren}
-      </div>
-    )
-  }
+function Input(props) {
+  const {
+    transparent, inverted, fluid,
+    icon, type, children, className,
+  } = props
+
+  // Semantic supports actions and labels on either side of an input.
+  // The element must be on the same side as the indicated class.
+  // We first determine the left/right classes for each type of child,
+  //   then we extract the children and place them on the correct side
+  //   of the input.
+  const isLeftAction = _.includes(className, 'left action')
+  const isRightAction = !isLeftAction && _.includes(className, 'action')
+  const isRightLabeled = _.includes(className, 'right labeled')
+  const isLeftLabeled = !isRightLabeled && _.includes(className, 'labeled')
+
+  const labelChildren = []
+  const actionChildren = []
+
+  Children.forEach(children, child => {
+    const isButton = child.type.name === 'Button'
+    const isDropdown = child.type.name === 'Dropdown'
+    // TODO: use child.type.name === 'Label' once Label component is merged.
+    const isLabel = _.isString(child.props.className) && !!child.props.className.match(/ui.*label$/)
+    const childIsAction = !isLabel && isButton || isDropdown
+
+    if (childIsAction) {
+      actionChildren.push(child)
+    } else if (isLabel) {
+      labelChildren.push(child)
+    }
+  })
+
+  const classes = cx('sd-input ui',
```

Comment:
Heads up, I've finished removing all `sd-*` classes, see #301.  Rebase to master to get the update.  The test for the `sd-*` class has been inverted to fail if any `sd-` class exists.

Let's remove all `sd-*` classes from all PRs.  👍